### ISSUE TITLE
Nothing renders

### ISSUE THREAD

**Issue by Garito** [DKirwan/calendar-heatmap#7]
Hi!
I'm trying to use your library but nothing renders
The selector disapears

Here my Javascript code:
```javascript
$('.calendar-heatmap').each(function(){
    var data = JSON.parse(JSON.parse($(this).data("data"))), corrected_data = [];
    for (var key in data) {
      corrected_data.push({date: Date.parse(key), count: data[key]});
    }

    this.heatmap = calendarHeatmap().data(corrected_data).selector(this);
    this.heatmap();
  });
```
Could you point me my mistake? or perhaps you need more info?

Thanks!

**Comment by Rsilnav** [comment]
Not knowing a lot, and not having tried this yet... It is necessary to have a double parse?
`JSON.parse(JSON.parse($(this).data("data")))`

**Comment by DKirwan** [comment]
Hey Garito, great to see that you're using the heatmap! Could you provide a link to a [JSFiddle](https://jsfiddle.net/) with the full code? I'd be glad to help out.

Rafael is right that the double `JSON.parse` may be unnecessary, but that will throw an error if that's the case.

**Comment by Garito** [comment]
The double JSON.parse is because I'm passing the data as a data attribute (see data-data="")
I can't create the JSFiddle right now (but I will) but I would like to explain the JSON thing
If I put a console.log showing the result of the repacking (the for) it shows the structure the help suggest (with a js date object as date) so I have to assume that there is no problem there

In case you have and idea about that

But I understand that my move will be to try this in the jsfiddle

Thanks by now

**Comment by DKirwan** [comment]
Thanks for clarifying. It sounds like the data is good, could be a problem with the selector perhaps.

I look forward to seeing the JSFiddle, I'll be able to give you a definite answer then :smile:

**Comment by Garito** [comment]
Could you point me how to include the library js's and css's without a CDN?
I'm trying to make the jsfiddle but don't know how to do this (I refus to include the whole css and js as text)

**Comment by Garito** [comment]
I found this:
https://rawgit.com/

**Comment by Garito** [comment]
What about this:
https://jsfiddle.net/hba77L4h/
Seems it does the same effect I'm issuing

**Comment by DKirwan** [comment]
Thanks, Garito. I reproduced your issue. When you draw or redraw the chart, it clears the element that contains the chart. Because you had a class `. calendar-heatmap` for the chart, the containing div got removed in error.

I've updated this to ensure it only removes the SVG element. 

The hotfix is now on the [master branch](https://github.com/DKirwan/calendar-heatmap/blob/master/src/calendar-heatmap.js), feel free to pull it and check it out! This defect was caused by an oversight on my part - thanks for pointing it out to me.
